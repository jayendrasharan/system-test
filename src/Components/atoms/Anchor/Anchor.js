import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const propTypes = {
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
  variation: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
  type: PropTypes.oneOf(['full-width', 'full-width-icon', '']),
  children: PropTypes.node,
  className: PropTypes.string,
  onClick: PropTypes.func,
  role: PropTypes.string,
};

export const createUrlSearchParams = query => {
  const queryValue = query || {};
  const queryParams = [];
  const keys = Object.keys(queryValue);
  for (let i = 0, l = keys.length; i < l; i += 1) {
    queryParams.push(`${keys[i]}=${queryValue[keys[i]]}`);
  }
  return queryParams.join('&');
};

const buildUrl = options => {
  if (typeof options === 'object') {
    const { pathname, query } = options;
    let url = pathname;
    if (typeof query === 'object') {
      url += `?${createUrlSearchParams(query)}`;
    }
    return url;
  }
  return options;
};


const Anchor = ({
  children,
  to,
  variation,
  type,
  className,
  onClick,
  role,
  ...others
}) => {
  return (
    <a
      role={role}
      href={buildUrl(to)}
      className={classnames('cta', type, variation, className)}
      onClick={onClick}
      data-hal-action-name={typeof children === 'string' && `text: ${children}`}
      {...others}
    >
      {children}
    </a>
  );
};

Anchor.propTypes = propTypes;
Anchor.defaultProps = {
  className: '',
  variation: 'primary',
  children: null,
  iconProps: null,
  isIconAfterContent: false,
  type: '',
  onClick: null,
  iconOnly: false,
  role: 'link',
};

export default Anchor;
