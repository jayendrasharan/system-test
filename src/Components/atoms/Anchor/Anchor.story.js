import React from 'react';
import { storiesOf } from '@storybook/react';

import { defaultAnchor, secondaryAnchor } from './Anchor.mock';
import Anchor from '.';

// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Atoms', module).addWithChapters('Anchor', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <Anchor
              {...defaultAnchor}
              className={`hide-default-sample ${defaultAnchor.className}`}
            >
              Home
            </Anchor>
          ),
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Primary Anchor Variations',
      sections: [
        {
          title: 'Default Link',
          sectionFn: () => <Anchor {...defaultAnchor}>Submit</Anchor>,
        },
        {
          title: 'Secondary Link',
          sectionFn: () => <Anchor {...secondaryAnchor}>Submit</Anchor>,
        },
      ],
    },
  ],
});
