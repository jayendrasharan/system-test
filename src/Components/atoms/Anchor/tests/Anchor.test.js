import React from 'react';
import { shallow } from 'enzyme';

import Anchor from '../Anchor';

describe('Anchor Component', () => {
  let anchor;

  test('should render correctly and match snapshot', () => {
    const props = {
      children: 'Sample',
      to: '/',
      onClick: () => {},
    };
    anchor = shallow(<Anchor {...props} />);
    expect(anchor).toMatchSnapshot();
  });
  test('should render correctly and match snapshot with object destination', () => {
    const props = {
      children: 'Sample',
      to: {
        pathname: 'http://abc.com',
        query: {
          a: 1,
          b: 2,
        },
      },
    };
    anchor = shallow(<Anchor {...props} />);
    expect(anchor).toMatchSnapshot();
  });
  test('should render correctly and match snapshot with object destination and no query param', () => {
    const props = {
      children: 'Sample',
      to: {
        pathname: 'http://abc.com',
        query: null,
      },
    };
    anchor = shallow(<Anchor {...props} />);
    expect(anchor).toMatchSnapshot();
  });
  test('should render correctly and match snapshot with icons after true', () => {
    const props = {
      children: 'Sample',
      to: '/',
      iconProps: {
        type: 'arrow-right',
        className: 'icon',
      },
      isIconAfterContent: true,
    };
    anchor = shallow(<Anchor {...props} />);
    expect(anchor).toMatchSnapshot();
  });
  test('should render correctly and match snapshot with icons after false', () => {
    const props = {
      children: 'Sample',
      to: '/',
      iconProps: {
        type: 'arrow-right',
        className: 'icon',
      },
      isIconAfterContent: false,
    };
    anchor = shallow(<Anchor {...props} />);
    expect(anchor).toMatchSnapshot();
  });
  test('should render correctly and match snapshot with icons after true', () => {
    const props = {
      children: 'Sample',
      to: '/',
      iconProps: {
        type: 'arrow-right',
        className: 'icon',
      },
      iconOnly: true,
      isIconAfterContent: true,
    };
    anchor = shallow(<Anchor {...props} />);
    expect(anchor).toMatchSnapshot();
  });
  test('should render correctly and match snapshot with icons after true', () => {
    const props = {
      children: 'Sample',
      to: '/',
      iconProps: {
        type: 'arrow-right',
        className: 'icon',
      },
      iconOnly: true,
      isIconAfterContent: false,
    };
    anchor = shallow(<Anchor {...props} />);
    expect(anchor).toMatchSnapshot();
  });
});
