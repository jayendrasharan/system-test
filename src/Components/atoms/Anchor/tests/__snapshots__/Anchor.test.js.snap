// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Component should render correctly and match snapshot 1`] = `
<a
  className="cta primary"
  data-hal-action-name="text: Sample"
  href="/"
  onClick={[Function]}
  role="link"
>
  Sample
</a>
`;

exports[`Anchor Component should render correctly and match snapshot with icons after false 1`] = `
<a
  className="cta primary"
  data-hal-action-name="text: Sample"
  href="/"
  onClick={null}
  role="link"
>
  <Icon
    className="icon mr1 align-middle arrow-right"
    focusable="false"
    height="18px"
    key="arrow-right"
    type="arrow-right"
    width="18px"
  />
  Sample
</a>
`;

exports[`Anchor Component should render correctly and match snapshot with icons after true 1`] = `
<a
  className="cta primary"
  data-hal-action-name="text: Sample"
  href="/"
  onClick={null}
  role="link"
>
  Sample
  <Icon
    className="icon ml1 align-middle arrow-right"
    focusable="false"
    height="18px"
    key="arrow-right"
    type="arrow-right"
    width="18px"
  />
</a>
`;

exports[`Anchor Component should render correctly and match snapshot with icons after true 2`] = `
<a
  className="cta primary"
  data-hal-action-name="text: Sample"
  href="/"
  onClick={null}
  role="link"
>
  Sample
  <Icon
    className="icon  align-middle arrow-right"
    focusable="false"
    height="18px"
    key="arrow-right"
    type="arrow-right"
    width="18px"
  />
</a>
`;

exports[`Anchor Component should render correctly and match snapshot with icons after true 3`] = `
<a
  className="cta primary"
  data-hal-action-name="text: Sample"
  href="/"
  onClick={null}
  role="link"
>
  <Icon
    className="icon  align-middle arrow-right"
    focusable="false"
    height="18px"
    key="arrow-right"
    type="arrow-right"
    width="18px"
  />
  Sample
</a>
`;

exports[`Anchor Component should render correctly and match snapshot with object destination 1`] = `
<a
  className="cta primary"
  data-hal-action-name="text: Sample"
  href="http://abc.com?a=1&b=2"
  onClick={null}
  role="link"
>
  Sample
</a>
`;

exports[`Anchor Component should render correctly and match snapshot with object destination and no query param 1`] = `
<a
  className="cta primary"
  data-hal-action-name="text: Sample"
  href="http://abc.com?"
  onClick={null}
  role="link"
>
  Sample
</a>
`;
